SMDS_3/include/CGAL/Simplicial_mesh_cell_base_3.h

```diff
@@ -32,187 +32,48 @@
 
 namespace CGAL {
 
-// Class Simplicial_mesh_cell_3
-// Cell base class used for tetrahedral remeshing
-// Adds information to Cb about the cell of the input complex containing it
-template< class Subdomain_index_,
-          class Surface_patch_index_,
-          class TDS>
+template <typename Gt,
+          typename SubdomainIndex,
+          typename SurfacePatchIndex,
+          typename Cb>
 class Simplicial_mesh_cell_3
+  : public Cb
 {
 public:
-  using Triangulation_data_structure = TDS;
-  using Vertex_handle = typename TDS::Vertex_handle;
-  using Cell_handle = typename TDS::Cell_handle;
-  using Vertex = typename TDS::Vertex;
-  using Cell = typename TDS::Cell;
-  using TDS_data = typename TDS::Cell_data;
+  using Vertex_handle = typename Cb::Vertex_handle;
+  using Cell_handle = typename Cb::Cell_handle;
+
+  using Geom_traits = Gt;
 
   // Index Type
-  using Subdomain_index = Subdomain_index_;
-  using Surface_patch_index = Surface_patch_index_;
+  using Subdomain_index = SubdomainIndex;
+  using Surface_patch_index = SurfacePatchIndex;
 
 public:
-  // Constructors
   Simplicial_mesh_cell_3()
     : time_stamp_(std::size_t(-1))
   {}
 
   Simplicial_mesh_cell_3(const Simplicial_mesh_cell_3& rhs)
-    : N(rhs.N)
-    , V(rhs.V)
-    , time_stamp_(rhs.time_stamp_)
-    , subdomain_index_(rhs.subdomain_index_)
+    : Cb(static_cast<const Cb&>(rhs)),
+      time_stamp_(rhs.time_stamp_),
+      subdomain_index_(rhs.subdomain_index_)
   {
-    for(int i=0; i <4; i++){
+    for(int i=0; i <4; ++i)
       surface_index_table_[i] = rhs.surface_index_table_[i];
```

Comment:
Even though it is just one assignment wouldn't it be better to use `{` / `}` as well?

Reply:
Note:
I see that the declarations for `surface_index_table_`  are all like:
```
std::array<Surface_patch_index, 4> surface_index_table_ = {};
```
and
```
Surface_patch_index surface_index_table_[4];
```
also the construct
```
CGAL_precondition(facet>=0 && facet<4);
```
is used in this context.

Wouldn't it make sense to have a `#define` (or alike) that defines this value of `4`?

Reply:
Thanks for the remarks.

>  Even though it is just one assignment wouldn't it be better to use { / } as well?
This code is old, and such updates would have to be mirrored into other packages. I don't want to extend the scope of this pull request.

> Wouldn't it make sense to have a #define (or alike) that defines this value of 4?
No, it's a 3D tetrahedral cell, there is no context in which it'll ever be something else than 4. I think it would be needlessly verbose to write:
```cpp
std::array<Surface_patch_index, NUMBER_OF_FACES_IN_A_TETRAHEDRON> surface_index_table_ = {};
```

Reply:
> > Even though it is just one assignment wouldn't it be better to use `{` / `}` as well?

> This code is old, and such updates would have to be mirrored into other packages. I don't want to extend the scope of this pull request.

It is a decisions that has to be taken on a higher level as I fully understand your reasoning. When doing it one place it should be consistent on all places (this is all a question of coding standards).

Indeed a tetrahedron has 4 faces and this will not change and the `NUMBER_OF_FACES_IN_A_TETRAHEDRON` would indeed be a bit long, From the name `surface_index_table_` it is not clear that this is a list of faces of a tetrahedron (might be my lack of knowledge of the CGAL package).